ably/options.go

```diff
@@ -1070,9 +1123,23 @@ func WithEchoMessages(echo bool) ClientOption {
 	}
 }
 
+// WithEndpoint is used for setting Endpoint using [ably.ClientOption].
+// Endpoint specifies the domain used to connect to Ably (REC1a).
+// If the endpoint option is specified then (REC1b):
+// If any one of the deprecated options environment, restHost, realtimeHost are also specified then the options as a set are invalid (REC1b1).
+// If the endpoint option is a domain name, determined by it containing at least one period (.) then the primary domain is the value of the endpoint option (REC1b2).
+// Otherwise, if the endpoint option is a non-production routing policy name of the form nonprod:[name] then the primary domain is [name].realtime.ably-nonprod.net (REC1b3).
+// Otherwise, if the endpoint option is a production routing policy name of the form [name] then the primary domain is [name].realtime.ably.net (REC1b4).
```

Comment:
Given this is a user facing docstring (i.e. it will appear [somewhere here](https://pkg.go.dev/github.com/ably/ably-go/ably#ClientOption)), I don't think it should have all this implementation detail and cryptic `(RECXX)` strings, that should be included as code comments within the implementation (e.g. on `clientOptions.getHostname()`), I think we want something simple and user friendly here, for example:

```suggestion
// WithEndpoint sets a custom endpoint for connecting to the Ably service
// (see [Platform Customization] for more information).
//
// [Platform Customization]: https://ably.com/docs/platform-customization
```

Reply:
I've updated this particular comment but we do seem to litter the comments with references to specs. Is it worth doing a more general tidy up of how we reference the specifications? I don't know we do it elsewhere, but when I first looked at the codebase I had no idea what they were referring to, and if a non-Ably person wanted to contribute, I guess they would have the same experience.

Rather than littering the code with references to specific spec codes, would it be better to just have a link to the overall spec for the version of the release?

Reply:
> Is it worth doing a more general tidy up of how we reference the specifications?

I think it is (although obviously not in this PR), since comments on exported types and methods are end user facing, and should be written with the end user in mind, no end user needs to see these spec point references, only SDK developers, so they should probably only appear within method definitions, and if there are any which end users might find useful, should probably be something like "For more information, see [REC1a](https://sdk.ably.com/builds/ably/specification/main/features/#REC1a) in the Ably SDK specification"